raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2154,8 +2182,14 @@ private boolean maybeCompleteShutdown(long currentTimeMs) {
         return false;
     }
 
-    private void maybeUpdateOldestSnapshotId() {
-        log.latestSnapshotId().ifPresent(log::deleteBeforeSnapshot);
+    private void maybeUpdateEarliestSnapshotId() {
```

Comment:
nit: it seems like the objective here is deleting old snapshots, which is why we call `deleteBeforeSnaphshot`. Updating the earliest snapshot id seems more like a side effect.

Reply:
Hmm, I guess I still see this the other way around. Why would the raft client care about updating the log start offset if not to delete old snapshots? What would that even mean outside the context of snapshot deletion?

Reply:
I changed the method name.